### PR TITLE
Support Debian 12 and refactor to make that easier

### PR DESCRIPTION
**Pull request by ekohl**
## Summary

This refactors the code to make it easier to support multiple versions without copying a lot of code.

## Related Issues

It's based on https://github.com/puppetlabs/puppetlabs-java/pull/569

## Checklist
- [x] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified. (For example `puppet apply`)

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]

File: metadata.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
       "operatingsystem": "Debian",
       "operatingsystemrelease": [
         "10",
-        "11"
+        "11",
+	"12"
```
[the remark this answers is not in the document]
Comment (smortex) [reply metadata.json]: Mixed space and tabs :wink:.  Not sure which plugin help me here, maybe "Refined Github"

screenshot of the above

[the remark this answers is not in the document]
Comment (smortex) [reply metadata.json]: BTW it is fixed in the final patch.

**Review by praj1001 — APPROVED** [review]

File: metadata.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ramesh7) [reply metadata.json]: The Debian 12 is not yet supported by Puppet eco-system, so just trying to grab thought that should we revert from metadata?

[the remark this answers is not in the document]
Comment (bastelfreak) [reply metadata.json]: Debian 12 packages were released a few days ago?

**Pull request merged**